### ISSUE TITLE
Text Set Labels

### ISSUE THREAD

**Issue by BrittanyIRL** [GoogleForCreators/web-stories-wp#6517]
## Feature Description

In the process of fixing text sets in the editor to be navigable by keyboard arrows, I was testing with a screen reader and realized we don't have any kind of labeling for text sets. Right now the text read is just `Insert Text Set` - there's no description of what it looks like. 

I'm curious if we want to add labeling to text sets. If yes, then we need that list before development can start. If no, then we can either leave the aria label as is or update it accordingly. 

## Additional Context

Perhaps some kind of label from the cross section of text set type and font as in figma? 
![Screen Shot 2021-02-24 at 11.20.25 AM.png](https://images.zenhubusercontent.com/5e9dcd9a9ff5dfdaffcdf132/ace02178-d0c8-4659-aaf4-385f7e1ae943)

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria

Following the below thread of comments we will do the following: 
- Label by text set section (ie List or Editorial) with number it is in the view starting with 1 and increasing as long as the text set is under the same grouping. 
- Labels are hidden and should be announced via screen reader and present in accessibility of dom.

**Comment by dreamofabear** [comment]
Probably a UX question but IMO "<category> <#>" could work e.g. "Editorial 3".

**Comment by swissspidy** [comment]
Previously mentioned in #6295

**Comment by swissspidy** [comment]
Also worth noting that we have the same situation with page layouts. This will become even more relevant once we allow creating custom layouts.

**Comment by BrittanyIRL** [comment]
> Previously mentioned in #6295

Yes, thank you for drawing attention to this. I set this ticket (#6517) as a blocker for #6295 when I made this ticket - it's the bug ticket I want to clear out and that's why I wanted us to talk about what we should label things. 😄 

> Also worth noting that we have the same situation with page layouts. This will become even more relevant once we allow creating custom layouts.

Great clarification, resounding **yes**. I'm hopeful that we can just establish a pattern, as I suggested in my original description above and as Will suggested in his comment. If that's the case then creating labels off the same pattern for page layouts and then also custom layouts wouldn't be too hard.

**Comment by o-fernandez** [comment]
I think that the suggestion above from Will makes sense. Since we don't seem to have a better suggestion, shall we go with that? @BrittanyIRL

**Comment by swissspidy** [comment]
> I think that the suggestion above from Will makes sense. Since we don't seem to have a better suggestion, shall we go with that? @BrittanyIRL

Now we just need a plan for page layouts...

**Comment by BrittanyIRL** [comment]
> > I think that the suggestion above from Will makes sense. Since we don't seem to have a better suggestion, shall we go with that? @BrittanyIRL
> 
> Now we just need a plan for page layouts...

Page layouts actually already has aria-labels (and labels on hover) that read out the page layout name. We've got this handy matrix from figma. 

<img width="276" alt="Screen Shot 2021-04-05 at 9 19 29 AM" src="https://user-images.githubusercontent.com/10720454/113597083-0d22b480-95f0-11eb-9c1a-9d7354f6b7d7.png">

**Comment by BrittanyIRL** [comment]
While implementing this and looking at what we do for page layouts (page templates), I thought perhaps doing something like this would be more useful for accessibility: 
`[text set category] with [fonts present in set] font(s)`

Examples: `cover text set with Roboto font`, `cover text set with Source Serif Pro, Lato fonts` 

For reference, page templates do this: `[template name] [template type]` 

Let me know if this is ok with you! @o-fernandez @choumx

**Comment by o-fernandez** [comment]
> While implementing this and looking at what we do for page layouts (page templates), I thought perhaps doing something like this would be more useful for accessibility:
> `[text set category] with [fonts present in set] font(s)`
> 
> Examples: `cover text set with Roboto font`, `cover text set with Source Serif Pro, Lato fonts`
> 
> For reference, page templates do this: `[template name] [template type]`
> 
> Let me know if this is ok with you! @o-fernandez @choumx

That sounds nice! 
Dumb question, but is this going to be coded in a way that works well for easy translations as well? (e.g., will the multiple permutations of the strings use translated terms as well?) May be a good question for our translations expert... @swissspidy

**Comment by BrittanyIRL** [comment]
it is! i already got his eyes on it!

**Comment by o-fernandez** [comment]
> it is! i already got his eyes on it!

Oh, excellent! then sounds great to me.

**Comment by BrittanyIRL** [comment]
@csossi acceptance for this has evolved a bit, you should get the below pattern for aria label on text sets: 
`Text set [text set category] with [fonts present in set] font(s)`, so like this: `text set Cover with Roboto font` or `text set Cover with Source Serif Pro, Lato fonts`.

**Comment by BrittanyIRL** [comment]
@csossi friendly ping on this, it's been waiting for qa for over 2 weeks now.

**Comment by csossi** [comment]
Verified in QA
